controllers/imagecollector/imagecollector_controller.go

```diff
@@ -222,6 +274,139 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	return ctrl.Result{RequeueAfter: *repeatPeriod}, nil
 }
 
+func (r *Reconciler) getChildImageJobs(ctx context.Context, collector *eraserv1alpha1.ImageCollector) ([]eraserv1alpha1.ImageJob, error) {
+	imageJobList := &eraserv1alpha1.ImageJobList{}
+	if err := r.List(ctx, imageJobList); err != nil {
+		log.Info("could not list imagejobs")
+		return nil, err
+	}
+
+	relevantJobs := util.FilterJobListByOwner(
+		imageJobList.Items, metav1.NewControllerRef(collector, schema.GroupVersionKind{
+			Group:   "eraser.sh",
+			Version: "v1alpha1",
+			Kind:    "ImageCollector",
+		}),
+	)
+
+	return relevantJobs, nil
+}
+
+func (r *Reconciler) upsertImageList(ctx context.Context, collector *eraserv1alpha1.ImageCollector) (ctrl.Result, error) {
+	imageListItems := make([]string, 0, len(collector.Status.Vulnerable))
+	images := collector.Status.Vulnerable
+
+	if *deleteScanFailedImages {
+		images = append(images, collector.Status.Failed...)
+	}
+
+	for i := range images {
+		img := images[i]
+		imageListItems = append(imageListItems, img.Digest)
+	}
+
+	imageList := eraserv1alpha1.ImageList{}
+	err := r.Get(ctx, types.NamespacedName{Namespace: "", Name: "imagelist"}, &imageList)
+	if !isNotFound(err) { // if *found*, then update
```

Comment:
is this a little confusing with the double negative?

Reply:
can we check if err is not nil instead?